### PR TITLE
Add dev server cli

### PR DESCRIPTION
**Pull request by cbourgois**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix 
| License          | BSD 3-Clause

## Description

Provide a cli to be able to run any folder in local with proxy to API.

In this PR: 
* Move proxy/sso config from `@ovh-ux/webpack-dev-server` to `@ovh-ux/dev-server-config` (and use it in `@ovh-ux/webpack-dev-server`)
* Add `@ovh-ux/dev-server` package that provide a cli to serve a path using proxy (allow custom `region`, `local2API`)

Example: 
```shell
$ cd /packages/manager/apps/hub
$ yarn build
...
$ cd /packages/manager/tools/dev-server
$ node bin/dev-server.js  ../../apps/hub/dist -p 1234

Serve: ../../apps/hub/dist on localhost:1234
```

## Todo

- [x] add `BREAKING CHANGE` in commits

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report test-manager_2#13295.0 ✘  
* Stage 1
  * e2e -   
  * lint ✔ 
  * SemVer ✘

**Comment by marie-j** [comment]
It could be nice to have documentation in the `README.md` with examples like what params are accepted, how you launch it etc etc :)

File: packages/manager/tools/dev-server-config/src/sso/index.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            console.error('[SSO] - crosslogin - error: ', err);
+            return resp.status(500);
+          }
+
+          const cookies = resp.headers['set-cookie'];
+
+          cookies.reverse().forEach((someCookie) => {
+            const parsedCookie = cookie.parse(someCookie);
+
+            if (parsedCookie['CA.OVH.SES']) {
```
Comment (marie-j) [inline packages/manager/tools/dev-server-config/src/sso/index.js:58]: Is this logic found that was written another way ?

File: packages/manager/tools/dev-server/bin/server.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    process.env.REGION || defaultConfig.region,
+  )
+  .option(
+    '-p, --port <port>',
+    'server port',
+    process.env.port || defaultConfig.port,
+  )
+  .option(
+    '--local2API',
+    'Use local2API proxy (localhost:8000)',
```
Comment (marie-j) [inline packages/manager/tools/dev-server/bin/server.js:22]:
```suggestion
    'Use local2API proxy (localhost:8080)',
```

**Comment by cbourgois** [comment]
> It could be nice to have documentation in the `README.md` with examples like what params are accepted, how you launch it etc etc :)

Added :)

File: packages/manager/tools/dev-server/bin/server.js (hunk 2 above)
Comment (cbourgois) [reply packages/manager/tools/dev-server/bin/server.js]: Fixed.

File: packages/manager/tools/dev-server-config/src/sso/index.js (hunk 1 above)
Comment (cbourgois) [reply packages/manager/tools/dev-server-config/src/sso/index.js:58]: Come from https://github.com/ovh/manager/blob/master/packages/manager/tools/webpack-dev-server/src/sso/index.ts#L63-L75

**Comment by cbourgois** [comment]
> Why some files are moved from `ts` to `js`, on webpack-dev-server ?

Some files are moved from `webpack-dev-server` to `dev-server-config` (from `ts`, to `js`). `dev-server-config` provide only the config to use in `dev-server` (using express) or `webpack-dev-server` (using ... `wepback`).
`webpack-dev-server` is still in `ts`, and the newcomer `dev-server-config` in `js`.

**Comment by JeremyDec** [comment]
@cbourgois  Yes I've seen this after, don't take care of my comment :)

File: packages/manager/tools/dev-server/package.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "@ovh-ux/dev-server",
+  "version": "0.0.0",
+  "description": "Provide OVHcloud Manager dev server",
+  "keywords": [
+    "codename",
```
Comment (antleblanc) [inline packages/manager/tools/dev-server/package.json:6]:
```suggestion
    "dev-server",
```

File: packages/manager/tools/dev-server/package.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  },
+  "dependencies": {
+    "commander": "^4.0.0",
+    "express": "^4.17.1",
+    "http-proxy-middleware": "^1.0.4"
+  },
+  "engines": {
+    "node": ">=8",
+    "yarn": ">=1.0.0"
+  }
```
Comment (antleblanc) [inline packages/manager/tools/dev-server/package.json:29]: **nitpick:** I'm not sure that we specifically want to support these engines versions.

I would suggest to be aligned with the existing [prerequisites](https://github.com/ovh/manager/#prerequisites).

```suggestion
```

File: packages/manager/tools/dev-server/src/default-config.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
```
Comment (antleblanc) [inline packages/manager/tools/dev-server/src/default-config.js:1]: **thought:** Probably can live into a `configs` folder isn't it ?

Ideas:
- `./src/configs/default.js`
- `./configs/default.js`

File: packages/manager/tools/dev-server-config/package.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  "main": "src/index.js",
+  "dependencies": {
+    "base64url": "^3.0.0",
+    "cookie": "^0.3.1",
+    "request": "^2.88.0"
+  },
+  "engines": {
+    "node": ">=8",
+    "yarn": ">=1.0.0"
+  }
```
[the remark this answers is not in the document]
Comment (cbourgois) [reply packages/manager/tools/dev-server-config/package.json:26]: Will update.
Come from https://github.com/ovh/manager/blob/master/packages/manager/tools/codename-generator/package.json#L27-L30

File: packages/manager/tools/dev-server-config/README.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+# Manager Dev Server Configuration
```
[the remark this answers is not in the document]
Comment (cbourgois) [reply packages/manager/tools/dev-server-config/README.md:1]: Added

**Comment by ovh-cds** [comment]
CDS Report build#13381.0 ✘  
* 
  * Build ✘

**Pull request merged**